src/Service.Tests/Caching/CachingConfigDeserializationTests.cs

```diff
@@ -0,0 +1,242 @@
+    [DataTestMethod]
+    public void GlobalCacheOptionsDeserialization_ValidJson(
+        string globalCacheConfig,
+        bool expectCacheEnabled)
+    {
+        // Arrange
+        string fullConfig = GetRawConfigJson(globalCacheConfig: globalCacheConfig, entityCacheConfig: string.Empty);
+
+        // Act
+        RuntimeConfigLoader.TryParseConfig(
+            json: fullConfig,
+            out RuntimeConfig? config,
+            logger: null,
+            connectionString: null,
+            replaceEnvVar: false,
+            dataSourceName: string.Empty,
+            datasourceNameToConnectionString: null,
+            replacementFailureMode: EnvironmentVariableReplacementFailureMode.Throw);
+
+        // Assert
+        Assert.IsNotNull(config, message: "Config must not be null, runtime config JSON deserialization failed.");
+        Assert.AreEqual(expected: expectCacheEnabled, actual: config.IsCachingEnabled, message: "RuntimeConfig.CacheEnabled expected to be: " + expectCacheEnabled);
+
+        if (expectCacheEnabled)
+        {
+            Assert.IsNotNull(config.Runtime?.CacheEnabled, message: "Expected global cache property to be non-null.");
+        }
+    }
+
+    /// <summary>
+    /// Validates that unexpected values provided for global runtime.cache property in the JSON config result in
+    /// a failure to deserialize the runtime config.
+    /// </summary>
+    /// <param name="globalCacheConfig">Escaped JSON string defining entity cache configuration.</param>
+    [DataRow(@",""cache"": {}", DisplayName = "Global cache property is empty: deserialization fails.")]
+    [DataTestMethod]
+    public void GlobalCacheOptionsDeserialization_InvalidJson(string entityCacheConfig)
+    {
+        // Arrange
+        string fullConfig = GetRawConfigJson(globalCacheConfig: entityCacheConfig, entityCacheConfig: string.Empty);
+
+        // Act
+        bool parsingSuccessful = RuntimeConfigLoader.TryParseConfig(
+            json: fullConfig,
+            out _,
+            logger: null,
+            connectionString: null,
+            replaceEnvVar: false,
+            dataSourceName: string.Empty,
+            datasourceNameToConnectionString: null,
+            replacementFailureMode: EnvironmentVariableReplacementFailureMode.Throw);
+
+        // Assert
+        Assert.IsFalse(parsingSuccessful, message: "Expected JSON parsing to fail.");
+    }
+
+    /// <summary>
+    /// Returns a JSON string of the runtime config with the test-provided
+    /// cache configuration.
+    /// </summary>
+    /// <param name="globalCacheConfig">Escaped JSON string defining global cache configuration.</param>
+    /// <param name="entityCacheConfig">Escaped JSON string defining entity cache configuration.</param>
+    /// <returns></returns>
+    private static string GetRawConfigJson(string globalCacheConfig, string entityCacheConfig)
+    {
+        StringBuilder expectedRuntimeConfigJson = new(
```

Comment:
is the StringBuilder needed? I would expected you could just return this directly as a string. Also, is it possible to remove the connection string or will that break the parsing in the test?

Reply:
With the multiple side effects that occur due to stringbuilder.Replace calls, I'd say it's better to have stringbuilder since you're not constantly copying/allocating strings for the test.

Reply:
also, yes i removed the connection string and all still works.